File: models/patient.py
"""
	Equivalent models component of an MVC
"""

class Patient:

	def __init__(self, attribute_dict):
		"""
			Attributes are in alphabetical order because dict type
			automatically sorts key values
		"""	
		self.allergies = attribute_dict["allergies"]	
		self.blood_type = attribute_dict["blood_type"]	
		self.date_of_birth = attribute_dict["date_of_birth"]	
		self.drugs_alchohol = attribute_dict["drugs_alchohol"]	
		self.first_name = attribute_dict["first_name"]	
		self.health_insurance = attribute_dict["health_insurance"]	
		self.height = attribute_dict["height"]	
		self.home_phone = attribute_dict["home_phone"]	
		self.last_name = attribute_dict["last_name"]
		self.middle_name = attribute_dict["middle_name"]	
		self.next_of_kin = attribute_dict["next_of_kin"]
		self.ssn = attribute_dict["ssn"]	
		self.vaccination = attribute_dict["vaccination"]
		self.vaccination_date = attribute_dict["vaccination_date"]
		self.sexually_active = attribute_dict["sexually_active"]
		self.weight = attribute_dict["weight"]
		self.work_phone = attribute_dict["work_phone"]

	# example methods
	def check_dob(self):
		pass

	
	def check_symptoms(self):
		pass

	def test_print(self):
		print("============================")	
		print("SSN: {}".format(self.ssn))
		print("First Name: {}".format(self.first_name))
		print("Middle Name: {}".format(self.middle_name))
		print("Last Name: {}".format(self.last_name))	
		print("============================")

File: criticare_app/lib/api/patient.dart
import 'package:http/http.dart' as http;
import '../models/patient.dart'; 
import 'dart:async';
import 'dart:convert';


class PatientApi {


    createPatient(Map<String, dynamic> formData) async {

        final _client = http.Client();
        var _loginUrl = Uri.parse('http://127.0.0.1:5000/patient');

        final response = await http.post(_loginUrl,
            headers: <String, String>{
                'Content-Type': 'application/json; charset=UTF-8',
            }, 
            body: jsonEncode({
                'ssn': formData['ssn'],
                'first_name': formData['firstName'], 
                'middle_name': formData['middleName'], 
                'last_name': formData['lastName'], 
                'date_of_birth': formData['dob'], 
                'height': formData['height'], 
                'weight': formData['weight'], 
                'next_of_kin': formData['nextOfKin'], 
                'home_phone': formData['homePhone'], 
                'work_phone': formData['workPhone'], 
                'symptoms': formData['symptoms'], 
                'health_insurance': formData['healthInsurance'], 
                'vaccination': formData['vaccination'], 
                'vaccination_date': formData['vaccinationDate'], 
                'drugs_alchohol': formData['drugsAlcohol'], 
                'sexually_active': formData['sexuallyActive'], 
                'allergies': formData['allergies'], 
                'blood_type': formData['bloodType'], 
                'notes': formData['notes']
            }),
        );
        //if this user gets a token, send them to the home page
        if (response.statusCode == 200) {
            print('Created Patient');
            return "Success"; 
        } else {
            return "Failure"; 
        }
    }


    static Future<Patient> fetchPatient(int? patientId) async {
        final response = await http.get(Uri.parse('http://127.0.0.1:5000/viewPatient?patient=${patientId}'));
        if (response.statusCode == 200) {
            // If the server did return a 200 OK response,
            // then parse the JSON.
            return Patient.fromJson(jsonDecode(response.body));
        } else {
            // If the server did not return a 200 OK response,
            // then throw an exception.
            throw Exception('Failed to load album');
        }
    }


    static Future<String> deletePatient(int? patientId) async {
        final response = await http.delete(Uri.parse('http://127.0.0.1:5000/patient?patient=${patientId}'));
        if (response.statusCode == 200) {
            // If the server did return a 200 OK response,
            // then parse the JSON.
            return "Deleted Patient"; 
        } else {
            // If the server did not return a 200 OK response,
            // then throw an exception.
            throw Exception('Failed to load album');
        }
    }


    static Future<List<Patient>> fetchPatients() async {
        final response = await http.get(Uri.parse('http://127.0.0.1:5000/patient'));
        if (response.statusCode == 200) {
            // If the server did return a 200 OK response,
            // then parse the JSON.
            final List<Patient> patients = patientFromJson(response.body);
            return patients;
        } else {
            // If the server did not return a 200 OK response,
            // then throw an exception.
            throw Exception('Failed to load album');
        }
    }

}